src/server/pps/server/api_server.go

```diff
@@ -95,12 +74,62 @@ func (a *apiServer) CreateJob(ctx context.Context, request *ppsclient.CreateJobR
 	if len(repoSet) < len(request.Inputs) {
 		return nil, fmt.Errorf("pachyderm.ppsclient.jobserver: duplicate repo in job")
 	}
+
+	jobID := uuid.NewWithoutDashes()
+
+	var parentJobInfo *persist.JobInfo
+	var err error
+	if request.ParentJob != nil {
+		inspectJobRequest := &ppsclient.InspectJobRequest{Job: request.ParentJob}
+		parentJobInfo, err = a.persistAPIServer.InspectJob(ctx, inspectJobRequest)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	reduce := false
+	if parentJobInfo != nil {
+		for _, jobInput := range parentJobInfo.Inputs {
+			if jobInput.Reduce {
+				reduce = true
+			}
+		}
+	}
```

Comment:
I'm a little confused about why we check `parentJobInfo.Inputs` rather than the `JobInfo.Inputs` to figure out if we're doing  `Reduce`. Was it this way before?